src/api/config.c

```diff
@@ -612,6 +612,49 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr)
+{
+    FILE *fPtr;
+    char buffer[256];
+    int foundSection = 0;
+    if (ParamPtr == NULL || SectionName == NULL || ParamName == NULL)
+        return M64ERR_INPUT_INVALID;
+    if (FileName == NULL || (fPtr = fopen(FileName, "rb")) == NULL)
+    {
+        DebugMessage(M64MSG_ERROR, "Unable to open config file '%s'.", FileName);
+        return M64ERR_INPUT_INVALID;
+    }
+    while (fgets(buffer, 255, fPtr) != NULL)
+    {
+        char *line = buffer;
+        ini_line l = ini_parse_line(&line);
+        switch (l.type)
+        {
+            case INI_SECTION:
+                if (osal_insensitive_strcmp(SectionName, l.name) == 0)
+                    foundSection = 1;
+                else
+                    foundSection = 0;
+                break;
+            case INI_PROPERTY:
+                if (foundSection)
+                {
+                    if (osal_insensitive_strcmp(ParamName, l.name) == 0)
+                    {
+                        fclose(fPtr);
+                        strncpy(ParamPtr, l.value, 256);
```

Comment:
PATH_MAX isn't always defined and since we need to check if it's defined or not, I think we better use an hardcoded value. I guess 4096 should be enough, this is the usual value of PATH_MAX on linux.